x-pack/plugins/security_solution/server/lib/detection_engine/rules/bulk_actions/rule_params_modifier.test.ts

```diff
@@ -51,26 +51,161 @@ describe('ruleParamsModifier', () => {
   });
 
   describe('index_patterns', () => {
-    test('should add new index pattern to rule', () => {
-      const editedRuleParams = ruleParamsModifier(ruleParamsMock, [
-        {
-          type: BulkActionEditType.add_index_patterns,
-          value: ['my-index-*'],
-        },
-      ]);
-      expect(editedRuleParams).toHaveProperty('index', ['initial-index-*', 'my-index-*']);
+    describe('add_index_patterns action', () => {
+      test.each([
+        [
+          '3 existing patterns + 2 of them = 3 patterns',
```

Comment:
= 5 patterns right? 🤔

Reply:
these scenarios dedicated to check that there is impossible to have duplicated values in index patterns.

So, if you have a, b, c patterns, and add a, b -> you end up with a, b, c (instead of a, b, c, a, b)

Similarly to other comments:   
>           '3 existing patterns - 2 new patterns = 3 patterns',

a, b, c - e, f = a, b, c
because we trying to remove 2 new patterns, that don't exist in a list

>           '3 existing patterns - 1 of them - 2 new patterns = 2 patterns',

a, b, c - c - e, f = a, b